frontend/src/components/crawl-list.ts

```diff
@@ -440,6 +412,40 @@ export class CrawlListItem extends LitElement {
     `;
   }
 
+  private renderActions() {
+    if (this.menuArr.length > 0) {
```

Comment:
You mentioned that this sometimes causes issues because `menuArr` isn't defined until after the first render — would it be worth reworking this component so that the list accepts an array of menu components as a prop, rather than as a slot?

Reply:
Quick fix could be:
```suggestion
    if (this.menuArr?.length > 0) {
```
And changing the type to:
```diff
- private menuArr!: Array<SlMenu>;
+ private menuArr?: Array<SlMenu>;
```

Reply:
Fixed in https://github.com/webrecorder/browsertrix-cloud/pull/1326/commits/5f8fb372edfd0d5a02627b3e3af229265902a61a

Reply:
Appears to work!  Thanks Sua!